packages/cosmos/src/client.rs

```diff
@@ -147,6 +148,131 @@ impl<Res> PerformQueryWrapper<Res> {
 }
 
 impl Cosmos {
+    async fn get_and_update_simulation_sequence(
+        &self,
+        address: Address,
+    ) -> Result<BaseAccount, Error> {
+        let mut guard = self.pool.get().await?;
+        let cosmos = guard.get_inner_mut();
+        let sequences = cosmos.simulate_sequences.read().await;
```

Comment:
Unless I'm mistaken: this is going to take a read guard for the rest of this function call, which will create a deadlock when we try to take the write guard below. This read guard should be dropped immediately.

Reply:
You're right. Have pushed a fix for it.